File: gex/src/model/HeadlessRunStatus.ts

export class HeadlessRunStatus {
    public gameID: string = "";
    public simulating: boolean = false;
    public frame: number = 0;
    public durationFrames: number = 0;
    public timestamp: Date = new Date();
    public fps: number = 0;

    public static parse(elem: any): HeadlessRunStatus {
        return {
            ...elem,
            timestamp: new Date(elem.timestamp)
        }
    }
}

File: gex/Models/Api/HeadlessRunStatus.cs
using System;

namespace gex.Models.Api {

	public class HeadlessRunStatus {

		/// <summary>
		///		ID of the game
		/// </summary>
		public string GameID { get; set; } = "";

		/// <summary>
		///		is the game still starting up (false), or is the game actually running? (true)
		/// </summary>
		public bool Simulating { get; set; }

		/// <summary>
		///		what frame the local simulation is on
		/// </summary>
		public long Frame { get; set; }

		/// <summary>
		///		how many frames will be simulated locally
		/// </summary>
		public long DurationFrames { get; set; }

		/// <summary>
		///		when this data was last updated
		/// </summary>
		public DateTime Timestamp { get; set; }

		/// <summary>
		///		how many frames per second are being processed
		/// </summary>
		public double Fps { get; set; }

	}
}
